Detached commands sometimes return results instead of id

### Description

```
DetachAttachITest.uptimePeriodically:63 » NumberFormat For input string: "Up 39 secs"
```

Seen randomly in the past, now it is sometimes more reproducible in #25574 , even locally, I just needed some 56 iterations until `DetachAttachITest#uptimePeriodically` failed.
When job finished and client did not disconnect yet, instead of sending him job ID, server sends real results of the job, which makes the response confusing and nonparseable.

It could be also asadmin bug, not sure now.

Expected behavior: when running detached command, always return the job id, then user uses attach to get results (however not progress).

This bug doesn't affect users, as they should not use detached commands for really fast commands.

Well, well, well, look what I have catched! Seems I have a reproducer.
The cause is that the `SseCommandHelper.invokeAsync` returns the detached job's eventOutput. When the job is faster than the job which started it, the output is taken from the async job, so user never receives the job id.
```
[ERROR] Errors: 
[ERROR]   DetachAttachITest.commandWithProgressStatus:79 » NumberFormat For input string: "Finished command process in 1 block(s)."
[ERROR]   DetachAttachITest.uptimePeriodically:63 » NumberFormat For input string: "Up 4 secs"

```

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -180,7 +180,6 @@ spec:
       S1AS_HOME = "${WORKSPACE}/glassfish7/glassfish"
       APS_HOME = "${WORKSPACE}/appserver/tests/appserv-tests"
       TEST_RUN_LOG = "${WORKSPACE}/tests-run.log"
-      GF_INTERNAL_ENV = credentials('gf-internal-env')
       PORT_ADMIN=4848
       PORT_HTTP=8080
       PORT_HTTPS=8181

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -26,26 +26,27 @@ def dumpSysInfo() {
    mvn -version || true
    ant -version || true
    ps -e -o start,etime,pid,rss,drs,command || true
-   cat /proc/cpuinfo || true
+   lscpu || true
    cat /proc/meminfo || true
    """
 }
 
 def startVmstatLogging(String stageName) {
    sh """
-   mkdir -p ${WORKSPACE}/logs
-   vmstat -t -w -a -y 1 > ${WORKSPACE}/logs/vmstat-${stageName}.log 2>&1 & echo \$! > ${WORKSPACE}/vmstat.pid
+   mkdir -p "${WORKSPACE}/logs"
+   vmstat -t -w -a -y 10 > "${WORKSPACE}/logs/vmstat-${stageName}.log" 2>&1 & echo \$! > "${WORKSPACE}/vmstat.pid"
    """
 }
 
-def stopVmstatLogging(String stageName) {
+def stopVmstatLogging() {
    sh """
-   if [ -f ${WORKSPACE}/vmstat.pid ]; then
-      pkill -F ${WORKSPACE}/vmstat.pid || true
-      rm -f ${WORKSPACE}/vmstat.pid
+   if [ -f "${WORKSPACE}/vmstat.pid" ]; then
+      pkill -F "${WORKSPACE}/vmstat.pid" || true
+      rm -f "${WORKSPACE}/vmstat.pid"
    fi
+   df -h || true
    """
-   archiveArtifacts artifacts: "logs/vmstat-${stageName}.log", allowEmptyArchive: true
+   archiveArtifacts artifacts: "logs/*", allowEmptyArchive: true
 }
 
 def antjobs = [
@@ -97,7 +98,7 @@ def generateAntPodTemplate(job) {
                   }
                }
             } finally {
-               stopVmstatLogging("ant-${job}")
+               stopVmstatLogging()
                archiveArtifacts artifacts: "${job}-results.tar.gz"
                junit testResults: 'results/junitreports/*.xml', allowEmptyResults: false
             }
@@ -222,7 +223,7 @@ spec:
                      mvn -B -e -fae clean validate -Ptck,set-version-id,staging
 
                      # Until we fix ANTLR in cmp-support-sqlstore, broken in parallel builds. Just -Pfast after the fix.
-                     mvn -B -e install -Pfastest,staging -T4C
+                     mvn -B -e install -Pfastest,staging,ci -T4C
                      ./gfbuild.sh archive_bundles
                      ./gfbuild.sh archive_embedded
 
@@ -233,7 +234,7 @@ spec:
                      '''
                   }
                } finally {
-                  stopVmstatLogging('mvn-build')
+                  stopVmstatLogging()
                }
                }
             }
@@ -255,11 +256,11 @@ spec:
                         dumpSysInfo()
                         timeout(time: 2, unit: 'HOURS') {
                            sh '''
-                           mvn -B -e clean install -Pstaging,qa
+                           mvn -B -e clean install -Pstaging,qa,ci
                            '''
                         }
                      } finally {
-                        stopVmstatLogging('mvn-tests')
+                        stopVmstatLogging()
                      }
                      }
                   }

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -28,6 +28,7 @@ def dumpSysInfo() {
    ps -e -o start,etime,pid,rss,drs,command || true
    lscpu || true
    cat /proc/meminfo || true
+   ulimit -a || true
    """
 }
 

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -147,10 +147,10 @@ spec:
     resources:
       limits:
         memory: "8Gi"
-        cpu: "6000m"
+        cpu: "5500m"
       requests:
         memory: "8Gi"
-        cpu: "6000m"
+        cpu: "5500m"
   volumes:
   - name: "jenkins-home"
     emptyDir:

--- a/appserver/tests/admin/tests/src/test/java/org/glassfish/main/admin/test/progress/DetachAttachITest.java
+++ b/appserver/tests/admin/tests/src/test/java/org/glassfish/main/admin/test/progress/DetachAttachITest.java
@@ -57,7 +57,7 @@ public class DetachAttachITest {
     @Test
     public void uptimePeriodically() throws Exception {
         final Set<String> ids = new HashSet<>();
-        for (int i = 0; i < 20; i++) {
+        for (int i = 0; i < 10; i++) {
             LOG.log(INFO, "detachAndAttachUptimePeriodically: round {0}", i);
             final String id;
             {

--- a/appserver/tests/admin/tests/src/test/java/org/glassfish/main/admin/test/progress/JobTestExtension.java
+++ b/appserver/tests/admin/tests/src/test/java/org/glassfish/main/admin/test/progress/JobTestExtension.java
@@ -95,7 +95,7 @@ public class JobTestExtension implements BeforeAllCallback, AfterAllCallback {
      */
     public static void waitForJobCleanup(long cleanupPeriod) {
         //  the configured period plus reserve time
-        final long maxEnd = System.currentTimeMillis() + cleanupPeriod * 1000L + 1000L;
+        final long maxEnd = System.currentTimeMillis() + cleanupPeriod * 1000L + 10000L;
         while (true) {
             AsadminResult result = ASADMIN.exec("list-jobs");
             assertThat(result, asadminOK());
@@ -114,7 +114,7 @@ public class JobTestExtension implements BeforeAllCallback, AfterAllCallback {
      */
     public static void waitForAllJobCompleted(long timeout) {
         //  the configured period plus reserve time
-        final long maxEnd = System.currentTimeMillis() + timeout * 1000L;
+        final long maxEnd = System.currentTimeMillis() + timeout * 10000L;
         while (true) {
             AsadminResult result = ASADMIN.exec("list-jobs");
             assertThat(result, asadminOK());

--- a/appserver/tests/gftest.sh
+++ b/appserver/tests/gftest.sh
@@ -18,7 +18,7 @@
 run_test() {
     local testid=${1}
     local found=false
-    
+
     for runtest in `find . -name run_test\.sh`; do
         for id in `${runtest} list_test_ids`; do
             if [[ "${id}" = "${testid}" ]]; then
@@ -34,7 +34,7 @@ run_test() {
             break
         fi
     done
-    
+
     if [[ "${found}" = false ]]; then
         echo Invalid Test Id.
         exit 1
@@ -42,14 +42,6 @@ run_test() {
 }
 
 if [ ! -z "${JENKINS_HOME}" ] ; then
-
-  # inject internal environment
-  readonly GF_INTERNAL_ENV_SH=$(mktemp -t XXXgf-internal-env)
-  if [ ! -z "${GF_INTERNAL_ENV}" ] ; then
-    echo "${GF_INTERNAL_ENV}" | base64 -d > ${GF_INTERNAL_ENV_SH}
-    . ${GF_INTERNAL_ENV_SH}
-    export MAVEN_OPTS="${MAVEN_OPTS} ${ANT_OPTS}"
-  fi
   export WSIMPORT_OPTS="${ANT_OPTS}"
 
   # setup the local repository

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateCommandDeleteResource.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateCommandDeleteResource.java
@@ -88,7 +88,7 @@ public class TemplateCommandDeleteResource extends TemplateExecCommand {
                     "Unable to parse the input entity. Please check the syntax.");
             throw new WebApplicationException(ResourceUtil.getResponse(400, /*parsing error*/ errorMessage, requestHeaders, uriInfo));
         }
-        return executeCommandAsSse(preprocessData(data));
+        return executeSseCommand(preprocessData(data));
     }
 
     //    //Handle POST request without any entity(input).

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateCommandGetResource.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateCommandGetResource.java
@@ -53,7 +53,7 @@ public class TemplateCommandGetResource extends TemplateExecCommand {
     @GET
     @Produces(SseFeature.SERVER_SENT_EVENTS + ";qs=0.5")
     public Response processSseGet() {
-        return executeCommandAsSse(prepareParameters());
+        return executeSseCommand(prepareParameters());
     }
 
     private ParameterMap prepareParameters() {

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateCommandPostResource.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateCommandPostResource.java
@@ -122,7 +122,7 @@ public class TemplateCommandPostResource extends TemplateExecCommand {
                     "Unable to parse the input entity. Please check the syntax.");
             throw new WebApplicationException(ResourceUtil.getResponse(400, /*parsing error*/ errorMessage, requestHeaders, uriInfo));
         }
-        return super.executeCommandAsSse(preprocessData(data));
+        return super.executeSseCommand(preprocessData(data));
     }
 
     @POST

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateExecCommand.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/TemplateExecCommand.java
@@ -28,11 +28,11 @@ import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.MultivaluedMap;
 import jakarta.ws.rs.core.PathSegment;
 import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.ResponseBuilder;
 import jakarta.ws.rs.core.Response.Status;
 import jakarta.ws.rs.core.UriInfo;
 
 import java.io.UnsupportedEncodingException;
-import java.net.HttpURLConnection;
 import java.net.URLDecoder;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -53,7 +53,6 @@ import org.glassfish.admin.rest.results.OptionsResult;
 import org.glassfish.admin.rest.utils.DetachedSseAdminCommandInvoker;
 import org.glassfish.admin.rest.utils.ResourceUtil;
 import org.glassfish.admin.rest.utils.SseAdminCommandInvoker;
-import org.glassfish.admin.rest.utils.SseEventOutput;
 import org.glassfish.admin.rest.utils.Util;
 import org.glassfish.admin.rest.utils.xml.RestActionReporter;
 import org.glassfish.api.ActionReport;
@@ -61,6 +60,9 @@ import org.glassfish.api.admin.CommandInvocation;
 import org.glassfish.api.admin.CommandRunner;
 import org.glassfish.api.admin.ParameterMap;
 import org.glassfish.internal.api.Globals;
+
+import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
+import static java.net.HttpURLConnection.HTTP_OK;
 
 /**
  * @author ludo
@@ -126,29 +128,23 @@ public class TemplateExecCommand extends AbstractResource implements OptionsCapa
         }
     }
 
-    protected final Response executeCommandAsSse(ParameterMap params) {
+    protected final Response executeSseCommand(ParameterMap params) {
         final boolean notify = params == null ? false : params.containsKey("notify");
         final boolean detach = params == null ? false : params.containsKey("detach");
         final CommandRunner<AdminCommandJob> commandRunner = Globals.getDefaultHabitat().getService(CommandRunner.class);
         final CommandInvocation<AdminCommandJob> invocation = commandRunner
             .getCommandInvocation(null, commandName, new RestActionReporter(), null, notify, detach).parameters(params);
-        final AsyncAdminCommandInvoker<SseEventOutput> invoker = detach
-            ? new DetachedSseAdminCommandInvoker(invocation)
-            : new SseAdminCommandInvoker(invocation);
-        final SseEventOutput output = invoker.start();
-        try {
-            return Response.status(HttpURLConnection.HTTP_OK).entity(output).build();
-        } finally {
-            if (detach) {
-                output.close();
-            }
-        }
+        final ResponseBuilder builder = Response.status(HTTP_OK);
+        final AsyncAdminCommandInvoker<Response> invoker = detach
+            ? new DetachedSseAdminCommandInvoker(new RestActionReporter(), invocation, builder)
+            : new SseAdminCommandInvoker(invocation, builder);
+        return invoker.start();
     }
 
     protected Response executeCommandLegacyFormat(ParameterMap data) {
         RestActionReporter actionReport = ResourceUtil.runCommand(commandName, data, getSubject());
         final ActionReport.ExitCode exitCode = actionReport.getActionExitCode();
-        final int status = (exitCode == ActionReport.ExitCode.FAILURE) ? HttpURLConnection.HTTP_INTERNAL_ERROR : HttpURLConnection.HTTP_OK;
+        final int status = exitCode == ActionReport.ExitCode.FAILURE ? HTTP_INTERNAL_ERROR : HTTP_OK;
         ActionReportResult option = optionsLegacyFormat();
         ActionReportResult results = new ActionReportResult(commandName, actionReport, option.getMetaData());
         results.getActionReport().getExtraProperties().putAll(option.getActionReport().getExtraProperties());

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/admin/CommandResource.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/admin/CommandResource.java
@@ -28,6 +28,7 @@ import com.sun.enterprise.util.StringUtils;
 import com.sun.enterprise.util.uuid.UuidGenerator;
 import com.sun.enterprise.util.uuid.UuidGeneratorImpl;
 import com.sun.enterprise.v3.admin.AdminCommandJob;
+import com.sun.enterprise.v3.admin.AsyncAdminCommandInvoker;
 import com.sun.enterprise.v3.common.ActionReporter;
 import com.sun.enterprise.v3.common.PlainTextActionReporter;
 import com.sun.enterprise.v3.common.PropsFileActionReporter;
@@ -59,8 +60,8 @@ import java.util.logging.Level;
 import javax.security.auth.Subject;
 
 import org.glassfish.admin.rest.RestLogging;
+import org.glassfish.admin.rest.utils.DetachedSseAdminCommandInvoker;
 import org.glassfish.admin.rest.utils.SseAdminCommandInvoker;
-import org.glassfish.admin.rest.utils.SseEventOutput;
 import org.glassfish.api.ActionReport;
 import org.glassfish.api.admin.CommandInvocation;
 import org.glassfish.api.admin.CommandModel;
@@ -73,6 +74,8 @@ import org.glassfish.hk2.api.ServiceLocator;
 import org.glassfish.internal.api.Globals;
 import org.glassfish.jersey.internal.util.collection.Ref;
 import org.glassfish.jersey.media.sse.SseFeature;
+
+import static java.net.HttpURLConnection.HTTP_OK;
 
 /**
  *
@@ -310,32 +313,27 @@ public class CommandResource {
         }
     }
 
-    private Response executeSseCommand(CommandName commandName, Payload.Inbound inbound, ParameterMap params, String modelETag,
-            Cookie jSessionId) throws WebApplicationException {
+    private Response executeSseCommand(CommandName commandName, Payload.Inbound inbound, ParameterMap params,
+        String modelETag, Cookie jSessionId) throws WebApplicationException {
         RestLogging.restLogger.log(Level.FINEST, "executeSseCommand({0})", commandName);
         final CommandModel model = getCommandModel(commandName);
         checkCommandModelETag(model, modelETag);
         final boolean notify = params == null ? false : params.containsKey("notify");
         final boolean detach = params == null ? false : params.containsKey("detach");
-        final CommandInvocation<AdminCommandJob> commandInvocation = getCommandRunner().getCommandInvocation(
+        final CommandInvocation<AdminCommandJob> invocation = getCommandRunner().getCommandInvocation(
             commandName.getScope(), commandName.getName(), new PropsFileActionReporter(), getSubject(), notify, detach);
         if (inbound != null) {
-            commandInvocation.inbound(inbound);
+            invocation.inbound(inbound);
         }
-        commandInvocation.outbound(new RestPayloadImpl.Outbound(false)).parameters(params);
-        ResponseBuilder rb = Response.status(HttpURLConnection.HTTP_OK);
+        invocation.outbound(new RestPayloadImpl.Outbound(false)).parameters(params);
+        ResponseBuilder builder = Response.status(HTTP_OK);
         if (isSingleInstanceCommand(model)) {
-            rb.cookie(getJSessionCookie(jSessionId));
+            builder.cookie(getJSessionCookie(jSessionId));
         }
-        final SseAdminCommandInvoker invoker = new SseAdminCommandInvoker(commandInvocation);
-        final SseEventOutput output = invoker.start();
-        try {
-            return rb.entity(output).build();
-        } finally {
-            if (detach) {
-                output.close();
-            }
-        }
+        final AsyncAdminCommandInvoker<Response> invoker = detach
+            ? new DetachedSseAdminCommandInvoker(new PropsFileActionReporter(), invocation, builder)
+            : new SseAdminCommandInvoker(invocation, builder);
+        return invoker.start();
     }
 
     private Response executeCommand(CommandName commandName, Payload.Inbound inbound, ParameterMap params, boolean supportsMultiparResult,
@@ -357,7 +355,7 @@ public class CommandResource {
         commandInvocation.outbound(outbound).parameters(params).execute();
         fixActionReporterSpecialCases(actionReporter);
         ActionReport.ExitCode exitCode = actionReporter.getActionExitCode();
-        int status = HttpURLConnection.HTTP_OK;
+        int status = HTTP_OK;
         if (exitCode == ActionReport.ExitCode.FAILURE) {
             status = HttpURLConnection.HTTP_INTERNAL_ERROR;
         }

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/utils/DetachedSseAdminCommandInvoker.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/utils/DetachedSseAdminCommandInvoker.java
@@ -18,6 +18,10 @@ package org.glassfish.admin.rest.utils;
 
 import com.sun.enterprise.v3.admin.AdminCommandJob;
 import com.sun.enterprise.v3.admin.AsyncAdminCommandInvoker;
+import com.sun.enterprise.v3.common.ActionReporter;
+
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.ResponseBuilder;
 
 import java.lang.System.Logger;
 
@@ -31,33 +35,39 @@ import static java.lang.System.Logger.Level.TRACE;
  * Starts the job, creates the {@link SseEventOutput} and writes the {@link ActionReport} containing
  * the job id as a message. Then ends.
  */
-public class DetachedSseAdminCommandInvoker extends AsyncAdminCommandInvoker<SseEventOutput> {
+public class DetachedSseAdminCommandInvoker extends AsyncAdminCommandInvoker<Response> {
     private static final Logger LOG = System.getLogger(DetachedSseAdminCommandInvoker.class.getName());
 
-    public DetachedSseAdminCommandInvoker(CommandInvocation<AdminCommandJob> commandInvocation) {
-        super(commandInvocation);
-    }
+    private final ActionReporter report;
+    private final ResponseBuilder builder;
 
+    public DetachedSseAdminCommandInvoker(ActionReporter idReport, CommandInvocation<AdminCommandJob> invocation, ResponseBuilder builder) {
+        super(invocation);
+        this.report = idReport;
+        this.builder = builder;
+    }
 
     /**
      * Starts the job, creates the {@link SseEventOutput} and writes the {@link ActionReport}
-     * containing
-     * the job id as a message. Then ends.
+     * containing the job id as a message. Then ends.
      */
     @Override
-    public SseEventOutput start() {
+    public Response start() {
         final AdminCommandJob job = getJob();
-        LOG.log(TRACE, "Job parameters: {0}, this: {1}", job.getParameters(), this);
-        final SseEventOutput eventOutput = new SseEventOutput(job);
-        LOG.log(TRACE, "Writing the job id. {0}", this);
-        final ActionReport report = job.getActionReport();
         report.setMessage(job.getId());
         report.setActionExitCode(ExitCode.SUCCESS);
-        // No other writes to the report or event output until we close.
-        report.lock();
-        eventOutput.write();
+        final Response response = createResponse(job);
+        LOG.log(TRACE, "Job parameters: {0}, this: {1}", job.getParameters(), this);
         startJob();
         LOG.log(TRACE, "Detached job started, leaving. {0}", this);
-        return eventOutput;
+        return response;
+    }
+
+    private Response createResponse(AdminCommandJob job) {
+        try (SseEventOutput eventOutput = new SseEventOutput(job)) {
+            LOG.log(TRACE, "Writing the job id. {0}", this);
+            eventOutput.write();
+            return builder.entity(eventOutput).build();
+        }
     }
 }

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/utils/SseAdminCommandInvoker.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/utils/SseAdminCommandInvoker.java
@@ -20,6 +20,9 @@ package org.glassfish.admin.rest.utils;
 import com.sun.enterprise.v3.admin.AdminCommandJob;
 import com.sun.enterprise.v3.admin.AsyncAdminCommandInvoker;
 
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.ResponseBuilder;
+
 import java.lang.System.Logger;
 
 import org.glassfish.api.admin.AdminCommandEventBroker;
@@ -36,15 +39,19 @@ import static org.glassfish.api.admin.AdminCommandState.EVENT_STATE_CHANGED;
 /**
  * Provides bridge between CommandInvocation and ReST Response for Server-Sent-Events.
  */
-public final class SseAdminCommandInvoker extends AsyncAdminCommandInvoker<SseEventOutput> {
+public final class SseAdminCommandInvoker extends AsyncAdminCommandInvoker<Response> {
     private static final Logger LOG = System.getLogger(SseAdminCommandInvoker.class.getName());
+
+    private final ResponseBuilder builder;
     private SseEventOutput eventOutput;
 
     /**
      * @param commandInvocation must not be null
+     * @param builder
      */
-    public SseAdminCommandInvoker(final CommandInvocation<AdminCommandJob> commandInvocation) {
+    public SseAdminCommandInvoker(final CommandInvocation<AdminCommandJob> commandInvocation, ResponseBuilder builder) {
         super(commandInvocation);
+        this.builder = builder;
     }
 
     /**
@@ -54,7 +61,7 @@ public final class SseAdminCommandInvoker extends AsyncAdminCommandInvoker<SseEv
      * @return {@link EventOutput} to be used for the communication with the client.
      */
     @Override
-    public SseEventOutput start() {
+    public Response start() {
         final AdminCommandJob job = getJob();
         LOG.log(TRACE, "Job parameters: {0}, this: {1}", job.getParameters(), this);
         this.eventOutput = new SseEventOutput(job);
@@ -68,7 +75,7 @@ public final class SseAdminCommandInvoker extends AsyncAdminCommandInvoker<SseEv
 
         LOG.log(TRACE, "Writing the current report and leaving. {0}", this);
         this.eventOutput.write();
-        return this.eventOutput;
+        return builder.entity(eventOutput).build();
     }
 
     private class ProgressEventListener implements AdminCommandListener<ProgressEvent>  {

--- a/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/utils/SseEventOutput.java
+++ b/nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/utils/SseEventOutput.java
@@ -79,9 +79,6 @@ public class SseEventOutput extends EventOutput {
             // and usually does when he receives the COMPLETED state.
             // The close() call does flush and may fail then.
             LOG.log(TRACE, "Failed to close the event output.", ex);
-        } finally {
-            // When we close, we abandon the lock so other threads can continue
-            job.getActionReport().unlock();
         }
     }
 }

--- a/nucleus/common/glassfish-api/src/main/java/org/glassfish/api/ActionReport.java
+++ b/nucleus/common/glassfish-api/src/main/java/org/glassfish/api/ActionReport.java
@@ -26,7 +26,6 @@ import java.util.List;
 import java.util.Map;
 import java.util.Properties;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.jvnet.hk2.annotations.Contract;
 
@@ -48,9 +47,8 @@ public abstract class ActionReport implements Serializable {
     private static final Logger LOG = System.getLogger(ActionReport.class.getName());
     private static final long serialVersionUID = -238144192513668688L;
 
-    private final AtomicBoolean locked = new AtomicBoolean(false);
     private Properties extraProperties;
-    private MessagePart topMessage = new MessagePart(locked);
+    private MessagePart topMessage = new MessagePart();
 
     public abstract void setActionDescription(String message);
 
@@ -74,16 +72,6 @@ public abstract class ActionReport implements Serializable {
 
     public abstract List<? extends ActionReport> getSubActionsReport();
 
-
-    public final void lock() {
-        LOG.log(DEBUG, "lock()");
-        this.locked.set(true);
-    }
-
-    public final void unlock() {
-        LOG.log(DEBUG, "unlock()");
-        this.locked.set(false);
-    }
 
     public MessagePart getTopMessagePart() {
         return topMessage;
@@ -218,27 +206,21 @@ public abstract class ActionReport implements Serializable {
 
         List<MessagePart> children = new ArrayList<>();
 
-        private final AtomicBoolean locked;
-
-        private MessagePart(AtomicBoolean locked) {
+        private MessagePart() {
             this.childrenType = "default";
-            this.locked = locked;
         }
 
         public MessagePart addChild() {
-            waitForUnlock();
-            MessagePart newPart = new MessagePart(locked);
+            MessagePart newPart = new MessagePart();
             children.add(newPart);
             return newPart;
         }
 
         public void setChildrenType(String type) {
-            waitForUnlock();
             this.childrenType = type;
         }
 
         public void setMessage(String message) {
-            waitForUnlock();
             if (this.message != null) {
                 LOG.log(DEBUG, () -> "Overwriting message '" + this.message + "' with '" + message + "'",
                     new RuntimeException());
@@ -249,7 +231,6 @@ public abstract class ActionReport implements Serializable {
         }
 
         public synchronized void appendMessage(String message) {
-            waitForUnlock();
             if (this.message == null) {
                 this.message = new StringBuilder(message);
             } else {
@@ -258,7 +239,6 @@ public abstract class ActionReport implements Serializable {
         }
 
         public void addProperty(String key, String value) {
-            waitForUnlock();
             props.put(key, value);
         }
 
@@ -324,12 +304,6 @@ public abstract class ActionReport implements Serializable {
                 }
             }
             return result.toString();
-        }
-
-        private void waitForUnlock() {
-            while (locked.get()) {
-                Thread.onSpinWait();
-            }
         }
 
         @Override

--- a/nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/AdminCommandJob.java
+++ b/nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/AdminCommandJob.java
@@ -23,7 +23,6 @@ import com.sun.enterprise.v3.admin.CheckpointHelper.CheckpointFilename;
 import java.io.File;
 import java.io.IOException;
 import java.io.ObjectInputStream;
-import java.io.Serializable;
 import java.lang.System.Logger;
 import java.util.List;
 import java.util.Objects;
@@ -53,7 +52,7 @@ import static java.lang.System.Logger.Level.WARNING;
  * @author Martin Mares
  * @author Bhakti Mehta
  */
-public final class AdminCommandJob implements Job, Serializable {
+public final class AdminCommandJob implements Job {
 
     private static final long serialVersionUID = 1L;
     private static final Logger LOG = System.getLogger(AdminCommandJob.class.getName());

--- a/nucleus/parent/pom.xml
+++ b/nucleus/parent/pom.xml
@@ -81,14 +81,10 @@
         <jakarta.validation-api.version>3.0.2</jakarta.validation-api.version>
         <hibernate-validator.version>8.0.2.Final</hibernate-validator.version>
 
-        <!-- Jakarta Web Services -->
-
         <!-- Jakarta XML Web Services -->
         <jakarta.xml.ws-api.version>4.0.2</jakarta.xml.ws-api.version>
 
-        <!-- Jakarta SOAP (with Attachments)
-       <jakarta.xml.soap-api.version>3.0.0-RC2</jakarta.xml.soap-api.version>
-        -->
+        <!-- Jakarta SOAP (with Attachments) -->
         <webservices.version>4.0.4</webservices.version>
 
         <!-- Jakarta Inject -->
@@ -119,9 +115,7 @@
         <jakarta.authorization-api.version>2.1.0</jakarta.authorization-api.version>
         <jakarta.authentication-api.version>3.0.0</jakarta.authentication-api.version>
 
-
         <!-- GlassFish Components -->
-
         <grizzly.version>4.0.2</grizzly.version>
         <grizzly.npn.version>2.0.0</grizzly.npn.version>
 
@@ -135,9 +129,7 @@
         <logging-annotation-processor.version>1.10</logging-annotation-processor.version>
         <command.security.maven.plugin.isFailureFatal>true</command.security.maven.plugin.isFailureFatal>
 
-
         <!-- 3rd party dependencies -->
-
         <commons-io.version>2.18.0</commons-io.version>
         <jboss.logging.annotation.version>3.0.4.Final</jboss.logging.annotation.version>
         <jboss.logging.version>3.6.1.Final</jboss.logging.version>
@@ -167,8 +159,12 @@
         <nucleus.install.dir.name>nucleus</nucleus.install.dir.name>
         <javadoc.skip>false</javadoc.skip>
         <deploy.skip>false</deploy.skip>
+
         <!-- Overrides the value from parent and enables the one we generate. -->
         <project.build.outputTimestamp>2025-04-22T16:51:49Z</project.build.outputTimestamp>
+
+        <!-- Show to developers by default -->
+        <checkstyle.consoleOutput>true</checkstyle.consoleOutput>
 
         <maven.test.jvmoptions.add-opens>--add-opens java.base/java.lang=ALL-UNNAMED</maven.test.jvmoptions.add-opens>
         <maven.test.jvmoptions.memory.sizes>-Xss512k -Xms256m -Xmx1g -XX:MaxDirectMemorySize=512m</maven.test.jvmoptions.memory.sizes>
@@ -948,7 +944,7 @@
                         <suppressionsLocation>org/glassfish/qa/config/checkstyle/checkstyle-suppressions.xml</suppressionsLocation>
                         <includeTestSourceDirectory>true</includeTestSourceDirectory>
                         <skip>${checkstyle.skip}</skip>
-                        <consoleOutput>true</consoleOutput>
+                        <consoleOutput>${checkstyle.consoleOutput}</consoleOutput>
                         <logViolationsToConsole>true</logViolationsToConsole>
                         <excludes>**/generated-sources/**/*, **/module-info.java</excludes>
                         <includes>**/*admin,**/*.adoc,**/*.bat,**/*.java,**/*.sh,**/*.xml</includes>
@@ -2097,21 +2093,11 @@
 
     <profiles>
         <profile>
-            <id>mac</id>
-            <activation>
-                <file>
-                    <exists>${java.home}/../Classes/classes.jar</exists>
-                </file>
-            </activation>
-            <dependencies>
-                <dependency>
-                    <groupId>com.sun</groupId>
-                    <artifactId>tools-jar</artifactId>
-                    <version>1</version>
-                    <scope>system</scope>
-                    <systemPath>${java.home}/../Classes/classes.jar</systemPath>
-                </dependency>
-            </dependencies>
+            <id>ci</id>
+            <properties>
+                <!-- Limit unuseful noise on CI while developers should see it -->
+                <checkstyle.consoleOutput>false</checkstyle.consoleOutput>
+            </properties>
         </profile>
         <profile>
             <id>oss-release</id>
@@ -2271,6 +2257,23 @@
             </build>
         </profile>
 
+        <profile>
+            <id>mac</id>
+            <activation>
+                <file>
+                    <exists>${java.home}/../Classes/classes.jar</exists>
+                </file>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>com.sun</groupId>
+                    <artifactId>tools-jar</artifactId>
+                    <version>1</version>
+                    <scope>system</scope>
+                    <systemPath>${java.home}/../Classes/classes.jar</systemPath>
+                </dependency>
+            </dependencies>
+        </profile>
         <profile>
             <id>windows</id>
             <activation>

